### PR TITLE
Cleanup csrlib

### LINKED ISSUE #2908: Clean up `csrlib`
From https://github.com/DMTF/libspdm/issues/2893#issuecomment-2484381881.

1. Clean up the header documentation so that it is accurate. 
2. Similar to `SET_CERTIFICATE`, add `is_busy` parameter to `libspdm_gen_csr`

### PR DESCRIPTION
**Pull request by steven-bellock**
Fix #2908.

This was tested with all four combination of values for `LIBSPDM_SET_CERT_CSR_PARAMS` and `LIBSPDM_ENABLE_CAPABILITY_CSR_CAP_EX`.

### CODE REVIEW

**Comment by steven-bellock** [comment]
@jyao1 I updated the `libspdm_gen_csr_ex` combination table so that it only includes combinations that libspdm will pass to the Integrator. Please review carefully.

@rw8896 to review as well.

File: os_stub/spdm_device_secret_lib_sample/csr.c [hunk 1]
```diff
@@ -327,7 +335,9 @@ bool libspdm_gen_csr_ex(
         if (*req_csr_tracking_tag == 0) {
             if (available_csr_tracking_tag == 0) {
                 /*no available tracking tag*/
-                *req_csr_tracking_tag = 0xFF;
+                #if LIBSPDM_SET_CERT_CSR_PARAMS
+                *is_busy = true;
```
Comment (rw8896) [inline os_stub/spdm_device_secret_lib_sample/csr.c:339]: The comment above said "no available tracking tag" so I am confused how does this imply "device is busy".
Would you help clarify this?

Comment (steven-bellock) [reply os_stub/spdm_device_secret_lib_sample/csr.c:339]: That comes from 
> If the Responder requires a reset to process a GET_CSR request, but does not have any available CSRTrackingTags, it shall respond with an ERROR message of ErrorCode=Busy.

**Review by rw8896 — COMMENTED** [review]
The change looks good to me.

**Comment by jyao1** [comment]
I am OK with that.

**Pull request merged**